var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -193,16 +203,58 @@ def flag_handler(self, name, flags):
 
         return (flags, None, None)
 
-    @run_before("install")
-    def set_blas_lapack(self):
-        # https://numpy.org/devdocs/user/building.html
-        # https://github.com/numpy/numpy/blob/master/site.cfg.example
+    def blas_lapack_pkg_config(self) -> Tuple[str, str]:
+        """Convert library names to pkg-config names.
 
-        # Skip if no BLAS/LAPACK requested
+        Returns:
+            The names of the blas and lapack libs that pkg-config should search for.
+        """
         spec = self.spec
-        if "+blas" not in spec and "+lapack" not in spec:
-            return
+        blas = spec["blas"].libs.names[0]
+        lapack = spec["lapack"].libs.names[0]
+
+        if spec["blas"].name in ["intel-mkl", "intel-parallel-studio", "intel-oneapi-mkl"]:
+            blas = "mkl-dynamic-lp64-seq"
```

Comment:
I'll note that we also have support for ILP64 now (and the wheels on PyPI use that), and also that `-omp` instead of `-seq` flavors should work. For the latter, perhaps the same logic as for `armpl` below is useful here? 

To enable the ILP64 support, use `-Csetup-args=-Duse-ilp64=true` in the `pip` invocation. SciPy doesn't yet support ILP64 though, so perhaps it's worth adding a comment for now but not implementing it, since this would prevent numpy and scipy from using the same BLAS/LAPACK library.

Reply:
I have pretty limited knowledge of MKL so I can't say that I fully understand what you're suggesting here. If you add code suggestions I can review them.

Reply:
I suggest to leave this open for now. It's easier to do in a follow-up, once I have `-Dblas=auto` and `-Dblas=mkl` working (should be done within the next ~3 weeks).

Reply:
By "leave this open" do you mean remove the override for MKL or leave it as is?

Reply:
leave as is